src/ui/SearchInterface/SearchInterface.ts

```diff
@@ -763,6 +778,49 @@ export class SearchInterface extends RootComponent implements IComponentBindings
     return QueryStateModel.defaultAttributes.quickview;
   }
 
+  private handleFacetValueState(stateToSet: { [key: string]: any }, fvFields: string[]): void {
+    const facetRef = BaseComponent.getComponentRef('Facet');
+    let fieldsWithoutFacets = [];
+    if (facetRef) {
+      const allFacets: Component[] = this.getComponents(facetRef.ID);
+      fieldsWithoutFacets = fvFields.filter(facetField => {
+        // Try to find a facet matching the `fv:` field state.
+        const stateKey = QueryStateModel.getFacetValueId(facetField);
+        const value = stateToSet[stateKey];
+        if (value && value.length > 0) {
+          const facetsWithField = allFacets.filter(facet => facet.options.field == facetField);
+          if (facetsWithField.length > 0) {
+            // We found a facet, remove the `fv:` and replace it with `f:`.
+            delete stateToSet[stateKey];
+            facetsWithField.forEach(facet => (stateToSet[QueryStateModel.getFacetId(facet.options.id)] = value));
+            return false;
+          }
+        }
+        return true;
+      });
+    } else {
+      fieldsWithoutFacets = fvFields;
+    }
+
+    // For the remaining field, we need to transform them in hidden queries.
+    // This ensure that an `fv:` state is always transformed into the filter it is supposed to apply.
+    if (fieldsWithoutFacets.length > 0) {
+      const valuesTransformedToHiddenQuery = fieldsWithoutFacets
+        .map(facetField => {
+          const stateKey = QueryStateModel.getFacetValueId(facetField);
+          const value = stateToSet[stateKey];
+          if (value && value.length > 0) {
+            delete stateToSet[stateKey];
+            return `${facetField}=="${value}"`;
+          }
+        })
+        .filter(expression => !!expression);
+      if (valuesTransformedToHiddenQuery.length > 0) {
+        stateToSet[QueryStateModel.attributesEnum.hq] = valuesTransformedToHiddenQuery.join(' AND ');
```

Comment:
This relies on the "CoveoHiddenQuery" always being present in the page to work. Meaning there is now a much more tight coupling between those two components for things to work properly.

Is there any way this could be encapsulated in a different manner ?

Maybe this new component and the hidden query component could share code so there's no repetition, but it would be better if this component could actually live on it's own in my opinion. As of now, it requires the orchestration of "FacetValueSuggestion", "Facets" and "HiddenQuery", with all three of them configured perfectly so that it works correctly.